core/network-libp2p/src/transport.rs

```diff
@@ -33,10 +33,26 @@ pub use self::bandwidth::BandwidthSinks;
 pub fn build_transport(
 	keypair: identity::Keypair
 ) -> (Boxed<(PeerId, StreamMuxerBox), io::Error>, Arc<bandwidth::BandwidthSinks>) {
+	// Build configuration objects for encryption mechanisms.
+	let noise_config = {
+		let noise_keypair = noise::Keypair::new().into_authentic(&keypair)
+			// For more information about this panic, see in "On the Importance of Checking
+			// Cryptographic Protocols for Faults" by Dan Boneh, Richard A. DeMillo,
+			// and Richard J. Lipton.
```

Comment:
A variant of this comment may be a good addition to [libp2p_core::identity::rsa::Keypair::sign](https://github.com/libp2p/rust-libp2p/blob/master/core/src/identity/rsa.rs#L55) since it primarily relates to RSA. Indeed, since substrate does not support RSA keypairs for libp2p (and does not plan to do so, I think), another option would just be to match on the keypair and panic here if the keypair is an RSA keypair, simply saying RSA is unsupported.

Reply:
> Indeed, since substrate does not support RSA keypairs for libp2p (and does not plan to do so, I think), another option would just be to match on the keypair and panic here if the keypair is an RSA keypair, simply saying RSA is unsupported.

You would still need the `expect()` though, since signing actually happens in `into_authentic`, so that's probably no improvement at all. Nevermind.